modules/broker-core/src/main/java/org/wso2/broker/core/Broker.java

```diff
@@ -37,12 +38,24 @@
 
     private final MessagingEngine messagingEngine;
 
+    private final AuthenticationManager authenticationManager;
+
     public Broker(StartupContext startupContext) throws Exception {
         this.messagingEngine = new MessagingEngine(startupContext.getService(DataSource.class));
 
         BrokerServiceRunner serviceRunner = startupContext.getService(BrokerServiceRunner.class);
         serviceRunner.deploy(new BrokerAdminService());
         startupContext.registerService(Broker.class, this);
+        this.authenticationManager = new AuthenticationManager(startupContext);
```

Comment:
Shall we read and pass the broker config here. It is better if we can have all the service reading logic in one place.